src/cart/crt_iv.c

```diff
@@ -2193,11 +2193,6 @@ handle_ivsync_response(const struct crt_cb_info *cb_info)
 		D_ASSERT(iv_sync->isc_ivns_internal == NULL);
 	}
 
-	if (iv_sync->isc_sync_type.ivs_comp_cb)
-		iv_sync->isc_sync_type.ivs_comp_cb(
-			iv_sync->isc_sync_type.ivs_comp_cb_arg,
-			cb_info->cci_rc);
-
 	if (iv_sync->isc_ivns_internal) {
```

Comment:
why are you removing code from crt_iv.c? i dont understand the rationale behind this

Reply:
oh this is originally added  for DAOS to releas the resources of asynchronous IV sync, but this patch will remove that use case, instead it will create extra ULT, then use EAGER IV sync.

The original implementation is a bit confusing on DAOS side,  so I remove this to make it cleaner.

Reply:
should this then instead be just conditional based on sync type? something like if (iv_sync->isc_sync_type.ivs_mode != LAZY) {  iv_sync->....

Reply:
hmm, we do not need this use case anyway.  So I just remove ivs_comp_cb() anyway. IMHO, probably create a ULT inside DAOS might seems cleaner.

Hmm, probably we do not need async IV in cart at all. I assume DAOS will be the only user for cart IV.

Reply:
we should remove option for it then. what i dont want to happen is to have part of the code (like headers) still mentioning eager, while we remove code that supported eager sync. or as i mentioned in the first comment we should keep it but make it conditional on eager, in case someone in future decided that we want to use async iv calls afterall

Reply:
IMHO, the async callback make it really confuse with the callback in crt_iv_update() parameters.

Since DAOS can do asynchronous easily with ULT, so I would rather make CART IV simpler.

Reply:
ah, you mean remove the whole LAZY part? yes, makes sense, I can do that in the following patch, if you are ok with that.

Reply:
ok